### PR TITLE
Injiweb 1622 : Refactor Crypto Utility Classes

### PR DESCRIPTION
**Pull request by Rudhhi-Shah-14**
renamed the files 
- CryptoCoreUtil to P12KeyStoreManager.
- CryptoUtil to  DerivedKeyCryptoUtil.

converted EncryptionDecryptionUtil to  DataProtectionService.

Replaced all the occurances of the old names with new ones

## Summary by CodeRabbit

* **Refactor**
  * Standardized and renamed cryptography services, consolidated keystore/key-management components, and reduced publicly injected crypto dependencies for more consistent, secure encryption flows.

* **Tests**
  * Expanded unit test coverage across key management, keystore loading, symmetric/asymmetric crypto, PIN-based flows, and error/edge cases to improve reliability and resilience.

### CODE REVIEW

File: src/test/java/io/mosip/mimoto/service/CredentialMatchingServiceTest.java [hunk 1]
```diff
@@ -42,7 +41,7 @@ public class CredentialMatchingServiceTest {
     private WalletCredentialService walletCredentialService;
 
     @Mock
-    private EncryptionDecryptionUtil encryptionDecryptionUtil;
+    private DataProtectionService dataProtectionService;
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/test/java/io/mosip/mimoto/service/CredentialMatchingServiceTest.java:44]: noted, leaving this as-is for now

File: src/test/java/io/mosip/mimoto/util/P12KeyStoreManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+    public void testLoadP12_InvalidFile() throws Exception {
+        java.io.File tempFile = java.io.File.createTempFile("invalid", ".p12");
+        tempFile.deleteOnExit();
+        java.io.FileWriter writer = new java.io.FileWriter(tempFile);
+        writer.write("Not a valid PKCS12 file");
+        writer.close();
+
+        KeyStore.PrivateKeyEntry entry = p12KeyStoreManager.loadP12(tempFile.getAbsolutePath(), "alias", "password");
+        assertNull(entry);
+        tempFile.delete();
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/test/java/io/mosip/mimoto/util/P12KeyStoreManagerTest.java]: acknowledged

File: src/test/java/io/mosip/mimoto/util/DerivedKeyCryptoUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        SecretKey key = new SecretKeySpec("testkey1234567890123456".getBytes(), AES_KEY_TYPE);
+        byte[] data = "testdata".getBytes();
+        byte[] aad = "testaad".getBytes();
+
+        InvocationTargetException exception = assertThrows(InvocationTargetException.class, () -> {
+            invokeSymmetricDecrypt(key, data, null, aad);
+        });
+
+        assertEquals(CryptoManagerException.class, exception.getTargetException().getClass());
+    }
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/test/java/io/mosip/mimoto/util/DerivedKeyCryptoUtilTest.java:158]: acknowledged

File: src/test/java/io/mosip/mimoto/util/DerivedKeyCryptoUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        byte[] data = "testdata".getBytes();
+        byte[] iv = "testiv".getBytes();
+        byte[] aad = "testaad".getBytes();
+
+        InvocationTargetException exception = assertThrows(InvocationTargetException.class, () -> {
+            invokeSymmetricDecrypt(key, data, iv, aad);
+        });
+
+        assertEquals(InvalidKeyException.class, exception.getTargetException().getClass());
+    }
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/test/java/io/mosip/mimoto/util/DerivedKeyCryptoUtilTest.java:145]: acknowledged

File: src/test/java/io/mosip/mimoto/util/DerivedKeyCryptoUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+            Throwable cause = ite.getTargetException().getCause();
+            boolean ok = cause instanceof NoSuchAlgorithmException || cause instanceof NoSuchPaddingException;
+            org.junit.jupiter.api.Assertions.assertTrue(ok, "Cause should be NoSuchAlgorithmException or NoSuchPaddingException");
+        } finally {
+            java.security.Security.removeProvider(bogus.getName());
+            for (java.security.Provider p : original) {
+                java.security.Security.addProvider(p);
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/test/java/io/mosip/mimoto/util/DerivedKeyCryptoUtilTest.java:178]: acknowledged

**Review by jackjain — APPROVED** [review]
LGTM

File: src/main/java/io/mosip/mimoto/util/WalletUtil.java [hunk 6]
```diff
@@ -28,11 +33,14 @@ public class WalletUtil {
...
+
+    @Autowired
+    private CryptomanagerService cryptomanagerService;
 
     public String decryptWalletKey(String encryptedWalletKey, String pin) {
         try {
-            return encryptionDecryptionUtil.decryptWithPin(encryptedWalletKey, pin);
+            return decryptWithPin(encryptedWalletKey, pin);
         } catch (Exception e) {
             throw new InvalidRequestException(INVALID_PIN.getErrorCode(), INVALID_PIN.getErrorMessage() + " " + e);
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/main/java/io/mosip/mimoto/util/WalletUtil.java]: acknowledged in the next commit

File: src/main/java/io/mosip/mimoto/util/WalletUtil.java [hunk 7]
```diff
@@ -103,4 +111,63 @@ public static void validateWalletId(HttpSession session, String walletIdFromRequ
...
+            throw new IllegalArgumentException("PIN cannot be null or empty");
+        }
+        try {
+            CryptoWithPinRequestDto requestDto = new CryptoWithPinRequestDto();
+            requestDto.setUserPin(pin);
+            String dataAsString = Base64.getEncoder().encodeToString(encryptionKey.getEncoded());
+            requestDto.setData(dataAsString);
+            CryptoWithPinResponseDto responseDto = cryptomanagerService.encryptWithPin(requestDto);
+            log.debug("Key encrypted with PIN successfully");
+            return responseDto.getData();
```
[the remark this answers is not in the document]
Comment (Rudhhi-Shah-14) [reply src/main/java/io/mosip/mimoto/util/WalletUtil.java:138]: Not necessary as both encryptWithPin and decryptWithPin in CryptomanagerService construct and return a CryptoWithPinResponseDto with non‑null data or throw exception on error, so the suggested null-checks are redundant.

**Review by jackjain — APPROVED** [review]
